api/src/modules/sections/sections.service.ts

```diff
@@ -1,33 +1,58 @@
+import { Repository, SelectQueryBuilder } from 'typeorm';
 import { AppBaseService } from '@api/utils/app-base.service';
 import { AppInfoDTO } from '@api/utils/info.dto';
-import { Injectable } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Section } from '@shared/dto/sections/section.entity';
-import { Repository, SelectQueryBuilder } from 'typeorm';
+import {
+  Section,
+  SectionWithDataWidget,
+} from '@shared/dto/sections/section.entity';
+import { FetchSpecification } from 'nestjs-base-service';
+import { ISurveyDataRepository } from '@api/infrastructure/survey-data-repository.interface';
+import { WidgetDataFiltersSchema } from '@shared/schemas/widget-data-filters.schema';
 
 @Injectable()
 export class SectionsService extends AppBaseService<
-  Section,
+  SectionWithDataWidget,
   unknown,
   unknown,
   AppInfoDTO
 > {
   public constructor(
     @InjectRepository(Section)
-    private sectionsRepository: Repository<Section>,
+    private readonly sectionsRepository: Repository<SectionWithDataWidget>,
+    @Inject('SurveyDataRepository')
+    private readonly surveyDataRepository: ISurveyDataRepository,
   ) {
     super(sectionsRepository, 'section', 'sections');
   }
 
-  // Does not allow me to use protected
+  // The library should make this method protected
   public async extendFindAllQuery(
-    query: SelectQueryBuilder<Section>,
-  ): Promise<SelectQueryBuilder<Section>> {
+    query: SelectQueryBuilder<SectionWithDataWidget>,
```

Comment:
This is a huge type mistake. The code that follows expects the widget to have a data property, but in the query that attaches it:

https://github.com/Vizzuality/4-growth/pull/119/files#diff-1c3686ea864ccbf37393991d95337ba2030e937bc4b913b144a3745fbf577252R56

We are skipping attaching the data property, when it is defined as mandatory.

Reply:
Removed

Reply:
If we leave as it was, we need to extends the AppBaseService like this:
![Uploading image.png…]()